pkg/apis/applicationnetworking/v1alpha1/accesslogpolicy_types.go

```diff
@@ -0,0 +1,65 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/aws/aws-application-networking-k8s/pkg/k8s"
+	"github.com/aws/aws-application-networking-k8s/pkg/model/core"
+)
+
+const (
+	AccessLogPolicyKind = "AccessLogPolicy"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:categories=gateway-api,shortName=tgp
+// +kubebuilder:storageversion
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+type AccessLogPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec AccessLogPolicySpec `json:"spec"`
+}
+
+// +kubebuilder:object:root=true
+// AccessLogPolicyList contains a list of AccessLogPolicies.
+type AccessLogPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []AccessLogPolicy `json:"items"`
+}
+
+// AccessLogPolicySpec defines the desired state of AccessLogPolicy.
+type AccessLogPolicySpec struct {
+	// The Amazon Resource Name (ARN) of the destination that will store access logs.
+	// Supported values are S3 Bucket, CloudWatch Log Group, and Firehose Delivery Stream ARNs.
+	//
+	// Changes to this value results in replacement of the VPC Lattice Access Log Subscription.
+	// +optional
+	DestinationArn *string `json:"protocol,omitempty"`
```

Comment:
Should be `json:"destinationarn,omitempty"`

 do we need add arn format validation logic in the `accesslogpolicy_types.go`  (by +kubebuilder:validation:Pattern=)?

Reply:
Set to destinationArn (camelCase) next commit.

Also added the format validation logic so that it matches what we have in the public API (Pattern=`^arn(:[a-z0-9]+([.-][a-z0-9]+)*){2}(:([a-z0-9]+([.-][a-z0-9]+)*)?){2}:([^/].*)?`)